core/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -144,4 +143,19 @@ protected static Object convertToStringIfBytesRef(Object obj) {
         }
         return queries;
     }
+
+    /**
+     * Utility method that converts inner query builders to xContent and
+     * checks for null values, rendering out empty object in this case.
+     */
+    protected static void doXContentInnerBuilder(XContentBuilder xContentBuilder, String fieldName,
+            QueryBuilder queryBuilder, Params params) throws IOException {
+        xContentBuilder.field(fieldName);
+        if (queryBuilder != null) {
+            queryBuilder.toXContent(xContentBuilder, params);
+        } else {
+            xContentBuilder.startObject();
```

Comment:
can we add a comment explaining why we do this. I think to mimic what QueryParseContext#parseInnerQueryBuilder accepts, which converts to null, so null goes back to `{}`. Is that right?